sdk/src/main/java/io/customer/sdk/CustomerIO.kt

```diff
@@ -182,6 +182,26 @@ class CustomerIO internal constructor(
         private var backgroundQueueSecondsDelay: Double =
             AnalyticsConstants.BACKGROUND_QUEUE_SECONDS_DELAY
 
+        fun setupConfig(config: Map<String, Any?>?): Builder {
+            if (config == null) return this
+
+            val logLevel = config.getProperty<String>(CustomerIOBuilderConfigKeys.Config.LOG_LEVEL).toCIOLogLevel()
+            setLogLevel(level = logLevel)
+            config.getProperty<String>(CustomerIOBuilderConfigKeys.Config.TRACKING_API_URL)?.takeIfNotBlank()?.let { value ->
+                setTrackingApiURL(value)
+            }
+            config.getProperty<Boolean>(CustomerIOBuilderConfigKeys.Config.AUTO_TRACK_DEVICE_ATTRIBUTES)?.let { value ->
+                autoTrackDeviceAttributes(shouldTrackDeviceAttributes = value)
+            }
+            config.getProperty<Int>(CustomerIOBuilderConfigKeys.Config.BACKGROUND_QUEUE_MIN_NUMBER_OF_TASKS)?.let { value ->
+                setBackgroundQueueMinNumberOfTasks(backgroundQueueMinNumberOfTasks = value)
+            }
+            config.getProperty<Double>(CustomerIOBuilderConfigKeys.Config.BACKGROUND_QUEUE_SECONDS_DELAY)?.let { value ->
```

Comment:
good point, still has to check RN, but for Flutter it works. 

regarding `string` i am not too sure, it might solve being flexible problem but typecasting is already scary, first, we will cast from `Any` to `String` and then to destined `Int/Double` increases the chances of mistakes also, open up questions, should booleans need to be string too then? 

So once we verify RN, we can do this in first iteration and then later on look into strings, if needed.

Reply:
@mrehan27 just checked I think RN is using `double` for everything instead of `int`. But `ReadableMap` does support `int` so not sure why can you double-check.

Reply:
@Shahroz16 You are correct, probably that's what I had in my mind when I added this comment 😅 IIRC, react converts `int` to `double` when passing it to `ReadableMap`, I'll re-confirm just to make sure it wasn't a mistake, but if not, we might have to support both these types.

Reply:
Just adding notes from our discussion, `ReadableMap` is flexible and supports both, but when converting to HashMap, only doubles are supported. So we'll be adding some work around to support both integers and doubles to make it work smoothly with React Native and Flutter.